test/secret/secret_test.go

```diff
@@ -0,0 +1,240 @@
+
+			// Download License File
+			licenseFilePath, err := testenv.DownloadFromS3Bucket()
+			Expect(err).To(Succeed(), "Unable to download license file")
+
+			// Create License Config Map
+			testenvInstance.CreateLicenseConfigMap(licenseFilePath)
+
+			// Create standalone Deployment with License Master
+			standalone, err := deployment.DeployStandaloneWithLM(deployment.GetName())
+			Expect(err).To(Succeed(), "Unable to deploy standalone instance with LM")
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			// Modify Secret key with new value
+			secretKey := testenv.SecretObject["HecToken"]
+			standalonePodName := fmt.Sprintf(testenv.StandalonePod, deployment.GetName(), 0)
+			HecToken := testenv.GetMountedKey(deployment, standalonePodName, secretKey)
+			modifiedHecToken := HecToken[:len(HecToken)-2] + "11"
+			testenv.ModifySecretKey(deployment, testenvInstance.GetName(), secretKey, modifiedHecToken)
+
+			// Ensure standalone is updating
+			testenv.VerifyStandalonePhase(deployment, testenvInstance, deployment.GetName(), splcommon.PhaseUpdating)
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			// Verify MC Pod is Ready
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			//Once system is up after update check each pod for secret key update
+			standaloneSecretName := fmt.Sprintf(testenv.SecretObjectPodName, deployment.GetName(), "standalone", 2)
+			licenseMasterSecretName := fmt.Sprintf(testenv.SecretObjectPodName, deployment.GetName(), "license-master", 2)
+			monitoringConsoleSecretName := fmt.Sprintf(testenv.SecretObjectPodName, testenvInstance.GetName(), "monitoring-console", 2)
+			verificationSecrets := []string{standaloneSecretName, licenseMasterSecretName, monitoringConsoleSecretName}
+
+			//Verify that each StatefulSet based secret has been update
+			updated := testenv.VerifySecretObjectUpdate(deployment, testenvInstance, verificationSecrets, secretKey, modifiedHecToken)
+			Expect(updated).To(Equal(true), "All keys not updated")
+
+			//All pods to be used to check for secret object update
+			licenseMasterPodName := fmt.Sprintf(testenv.LicenseMasterPod, deployment.GetName(), 0)
+			monitoringConsolePodName := fmt.Sprintf(testenv.MonitoringConsolePod, testenvInstance.GetName(), 0)
+			verificationPods := []string{standalonePodName, licenseMasterPodName, monitoringConsolePodName}
+
+			//Verify that new token is mounted on each pod
+			updated = testenv.VerifySecretPodUpdate(deployment, testenvInstance, verificationPods, secretKey, modifiedHecToken)
```

Comment:
VerifySecretPodUpdate should fail the test case. Thats the pattern which is used across the framework. In your current implementation of VerifySecretPodUpdate its not a verification rather a check.

Reply:
Modified